fury/primitive.py

```diff
@@ -117,18 +117,19 @@ def repeat_primitive(vertices, faces, centers, directions=(1, 0, 0),
         Expanded triangles that composed our shape to duplicate
     big_colors : ndarray
         Expanded colors applied to all vertices/faces
-    big_centers : ndarray
-        Expanded centers for all vertices/faces
 
     """
     # duplicated vertices if needed
     if not have_tiled_verts:
         vertices = np.tile(vertices, (centers.shape[0], 1))
-    big_vertices = vertices
+
     # Get unit shape
     unit_verts_size = vertices.shape[0] // centers.shape[0]
     unit_triangles_size = faces.shape[0]
 
+    big_centers = np.repeat(centers, unit_verts_size, axis=0)
+    # apply centers position
+    big_vertices = vertices + big_centers
```

Comment:
Same questions as above, I don't understand why you add this part? Please, can you explain ?